tests/test_administrative_borders.py

```diff
@@ -1,16 +1,149 @@
+import os
 import yaml
+from pathlib import Path
 
 import geopandas as gpd
 import pytest
+import shapely.geometry
+import pycountry
+import fiona
 
-# Loading the 20M file for its smaller size (1M used in actual workflow)
-URL_NUTS = "https://ec.europa.eu/eurostat/cache/GISCO/distribution/v2/nuts/shp/NUTS_RG_20M_{}_4326.shp.zip"
+from src.administrative_borders import _study_area, _to_multi_polygon, _drop_countries, _drop_geoms
+from src.conversion import eu_country_code_to_iso3
+# Loading the 60M file for its smaller size (1M used in actual workflow)
+URL_NUTS = "https://ec.europa.eu/eurostat/cache/GISCO/distribution/v2/nuts/shp/NUTS_RG_60M_{}_4326.geojson"
+ROOT_DIR = Path(os.path.abspath(__file__)).parent.parent
+PATH_TO_BORDERS = ROOT_DIR / "build" / "administrative-borders.gpkg"
 
 with open('config/schema.yaml', 'r') as src:
     config_schema = yaml.safe_load(src)
 
+with open('config/default.yaml', 'r') as src:
+    config_default = yaml.safe_load(src)
+
+@pytest.fixture
+def config(exclusions=0):
+    _config = {
+        "scope": {
+            'bounds': {
+                'x_min': 0,
+                'x_max': 1,
+                'y_min': 50,
+                'y_max': 51,
+            }
+        }
+    }
+    if exclusions > 0:
+        _config["scope"]["exclusion_zones"] = {
+            'foo': {
+                'x_min': 0.5,
+                'x_max': 0.6,
+                'y_min': 50.2,
+                'y_max': 50.5,
+            }
+        }
+    if exclusions == 2:
+        _config["scope"]["exclusion_zones"]["bar"] = {
+            'x_min': 0,
+            'x_max': 0.1,
+            'y_min': 50,
+            'y_max': 50.1,
+        }
+    return _config
+
+@pytest.fixture
+def europe_gdf():
+    return gpd.read_file(
+        URL_NUTS.format(config_schema["properties"]["parameters"]["properties"]["nuts-year"]["default"])
```

Comment:
since we should depend more on the schema for defaults, I'd say; it's more likely to remain static, no?

Reply:
Actually, having two locations with default values is confusing, isn't it? Maybe the schema simply should not contain defaults?

Because `defaults.yaml` is hard-coded into the Snakemake workflow it acts as a default anyways! If you do `snakemake --configfile my-weird-config.yaml`, Snakemake always falls back to `default.yaml` when a parameter is missing.

Conclusion: I'd rather not have defaults in the schema.

Reply:
OK, I think it can be a benefit for a. having the default information right next to the available parameters and b. knowing which parameters are essential to workflow operation (they have a default) and which aren't. Saying that, I'm happy to stick with defaults only in default.yaml.